src/coreclr/jit/llvm.cpp

```diff
@@ -1686,10 +1874,29 @@ void Llvm::ConvertShadowStackLocals()
                 CORINFO_SIG_INFO calleeSigInfo;
                 _compiler->eeGetMethodSig(callNode->gtCallMethHnd, &calleeSigInfo);
 
-                if (needsReturnStackSlot(calleeSigInfo.retType, calleeSigInfo.retTypeClass))
+                if (callNode->gtCallArgs != nullptr)
+                {
+                    // TODO-LLVM: out args?  E.g.
+                    //    /--*  t398   int    arg4 out+10
+                    //    +--*  t399   int    arg5 out + 14
+                    //    +--*  t908   byref  arg6 out + 18
+                    //    +--*  t397   int    arg3 in r9
+                    //    +--*  t395   ref    arg2 in r8
+                    //    +--*  t392   byref  arg0 in rcx
+                    //    +--*  t393   ref    arg1 in rdx
+
+                    // We come here also for ctors like the following
+                    //                                         /--*  t18    int    arg0 in rcx
+                    // N003(17, 5)[000019]-- CXG-- -----t19 = *CALL ref System.String..ctor
+                    failFunctionCompilation();
+                }
+
+                var_types callReturnType = callNode->TypeGet();
+                // ctors in CORINFO_SIG_INFO are void, but in IR are TYP_REF.  ILtoLLVMImporter does not have this problem as it uses MethodSignature
+                // which has the return type.  TODO-LLVM: is it better to create a new EE callback that matches IR/MethodSignature regards return types for ctors?
+                if (callReturnType == TYP_REF || needsReturnStackSlot(calleeSigInfo.retType, calleeSigInfo.retTypeClass))
```

Comment:
Thanks, I'll do this one and leave the other calls to `needsReturnStackSlot` as is because I think some or all of them will go when the args are lowered.